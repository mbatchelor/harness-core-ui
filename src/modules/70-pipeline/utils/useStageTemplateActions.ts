/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { defaultTo } from 'lodash-es'
import { parse } from 'yaml'
import produce from 'immer'
import { useCallback } from 'react'
import type { StageElementConfig } from 'services/cd-ng'
import { usePipelineContext } from '@pipeline/components/PipelineStudio/PipelineContext/PipelineContext'
import { createTemplate, getStageType } from '@pipeline/utils/templateUtils'
import type { TemplateSummaryResponse } from 'services/template-ng'

interface TemplateActionsReturnType {
  addOrUpdateTemplate: (selectedTemplate?: TemplateSummaryResponse) => Promise<void>
  removeTemplate: () => Promise<void>
}

export function useStageTemplateActions(): TemplateActionsReturnType {
  const {
    state: {
      selectionState: { selectedStageId = '' },
      templateTypes
    },
    updateStage,
    getStageFromPipeline,
    getTemplate
  } = usePipelineContext()
  const { stage } = getStageFromPipeline(selectedStageId)

  const addOrUpdateTemplate = useCallback(
    async (selectedTemplate?: TemplateSummaryResponse) => {
      try {
        const { template, isCopied } = await getTemplate({
          templateType: 'Stage',
          selectedChildType: getStageType(stage?.stage, templateTypes),
          selectedTemplate
        })
        const node = stage?.stage
        const processNode = isCopied
          ? produce(defaultTo(parse(template?.yaml || '')?.template.spec, {}) as StageElementConfig, draft => {
              draft.name = defaultTo(node?.name, '')
              draft.identifier = defaultTo(node?.identifier, '')
            })
          : createTemplate(node, template)
        await updateStage(processNode)
      } catch (_) {
        // Do nothing.. user cancelled template selection
      }
    },
    [stage?.stage, templateTypes, getTemplate, updateStage]
  )

  const removeTemplate = useCallback(async () => {
    const node = stage?.stage
    const processNode = produce({} as StageElementConfig, draft => {
      draft.name = defaultTo(node?.name, '')
      draft.identifier = defaultTo(node?.identifier, '')
      draft.type = getStageType(node, templateTypes)
    })
    await updateStage(processNode)
  }, [stage?.stage, templateTypes, updateStage])

  return { addOrUpdateTemplate, removeTemplate }
}
