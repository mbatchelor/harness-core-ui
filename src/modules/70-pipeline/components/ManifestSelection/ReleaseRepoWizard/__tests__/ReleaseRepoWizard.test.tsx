/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { findByText, fireEvent, render } from '@testing-library/react'

import { TestWrapper } from '@common/utils/testUtils'

import { ManifestDataType, manifestStoreTypes } from '../../Manifesthelper'
import ReleaseRepoWizard from '../ReleaseRepoWizard'

const props = {
  types: [
    ManifestDataType.K8sManifest,
    ManifestDataType.Values,
    ManifestDataType.HelmChart,
    ManifestDataType.OpenshiftTemplate,
    ManifestDataType.OpenshiftParam,
    ManifestDataType.Kustomize,
    ManifestDataType.KustomizePatches
  ],
  manifestStoreTypes: manifestStoreTypes,
  labels: {
    firstStepName: 'test1',
    secondStepName: 'test2'
  },
  selectedManifest: null,
  newConnectorView: false,
  handleConnectorViewChange: jest.fn(),
  handleStoreChange: jest.fn(),
  manifest: null,
  isReadonly: false,
  stage: {
    stage: {
      name: 'Stage Name',
      identifier: 'stage_id',
      spec: {
        serviceConfig: {
          serviceDefinition: {
            spec: {},
            type: 'Kubernetes'
          },
          serviceRef: 'knkjm'
        }
      }
    }
  },
  updateStage: jest.fn(),
  onClose: jest.fn(),
  initialValues: { connectorRef: undefined, selectedManifest: null, store: 'Git' }
}
describe('Release Repo wizard tests', () => {
  test('initial render', () => {
    const { container } = render(
      <TestWrapper>
        <ReleaseRepoWizard {...props} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test(`new connector view works correctly`, async () => {
    const { container } = render(
      <TestWrapper>
        <ReleaseRepoWizard {...props} />
      </TestWrapper>
    )

    const gitconnectorCard = container.getElementsByClassName('Thumbnail--squareCardContainer')[0]
    fireEvent.click(gitconnectorCard)
    const newConnectorLabel = await findByText(container, 'newLabel Git connector')
    expect(newConnectorLabel).toBeDefined()
    const newConnectorBtn = container.getElementsByClassName('addNewManifest')[0]
    expect(newConnectorBtn).toBeDefined()
    fireEvent.click(newConnectorLabel)

    expect(container).toMatchSnapshot()
  })
})
