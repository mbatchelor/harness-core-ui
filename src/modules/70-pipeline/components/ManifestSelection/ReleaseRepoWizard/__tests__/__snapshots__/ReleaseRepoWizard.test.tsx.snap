// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Release Repo wizard tests initial render 1`] = `
<div>
  <div
    class="StepWizard--main manifestWizard StepWizard--navBar"
  >
    <div
      class="StepWizard--navBarList"
    >
      <div
        class="StepWizard--navStep StepWizard--activeStep"
      >
        <span
          class="StepWizard--number"
        >
          1
        </span>
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main StepWizard--stepName"
          style="--text-line-clamp: 2; width: 200px;"
        >
          Release Repo Store
        </p>
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main StepWizard--stepName"
          style="--text-line-clamp: 2; width: 200px;"
        >
          
        </p>
      </div>
      <div
        class="StepWizard--navStep"
      >
        <span
          class="StepWizard--number"
        >
          2
        </span>
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main StepWizard--stepName"
          style="--text-line-clamp: 2; width: 200px;"
        >
          RepoDetails
        </p>
      </div>
    </div>
    <div
      class="StepWizard--stepDetails"
    >
       
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main optionsViewContainer"
        style="height: inherit;"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-variation-h3 StyledProps--margin-bottom-medium"
        >
          Release Repo Store
        </p>
        <div
          class="OverlaySpinner--overlaySpinner"
        >
          <form
            action="#"
            class="FormikForm--main"
          >
            <div>
              <div
                class="Layout--vertical StyledProps--main manifestForm StyledProps--flex StyledProps--flex-justifyContent-space-between StyledProps--flex-alignItems-flex-start"
              >
                <div
                  class="Layout--vertical StyledProps--main"
                >
                  <div
                    class="Layout--horizontal Layout--layout-spacing-large StyledProps--main"
                  >
                    <div
                      class="bp3-form-group ThumbnailSelect--mainContainer thumbnailSelect"
                      data-id="store-0"
                    >
                      <div
                        class="bp3-form-content"
                      >
                        <div
                          class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
                        >
                          <label
                            class="Thumbnail--squareCardContainer"
                          >
                            <div
                              class="bp3-card bp3-elevation-0 Card--card Thumbnail--squareCard Card--selected"
                            >
                              <span
                                class="Card--corner"
                              >
                                <span
                                  class="Card--badge"
                                />
                                <span
                                  class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-white"
                                  data-icon="main-tick"
                                >
                                  <svg
                                    height="10"
                                    viewBox="0 0 11 8"
                                    width="10"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M9.219.894c.356-.38.982-.413 1.398-.072.391.32.457.857.167 1.236l-.06.07-5.138 5.486c-.362.386-.993.41-1.403.068l-.07-.064-3.2-3.214A.88.88 0 01.95 3.098a1.045 1.045 0 011.328-.003l.074.067 2.443 2.454L9.22.894z"
                                      fill="currentColor"
                                    />
                                  </svg>
                                </span>
                              </span>
                              <span
                                class="bp3-icon StyledProps--main"
                                data-icon="service-github"
                              >
                                <svg
                                  height="26"
                                  viewBox="0 0 14 14"
                                  width="26"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M13.6 6.4l-6-6C7.3.1 6.7.1 6.4.4L5.1 1.6l1.6 1.6c.4-.1.8 0 1.1.3s.4.7.2 1.1l1.5 1.5c.4-.1.8 0 1.1.2.4.4.4 1.1 0 1.5s-1.1.4-1.5 0c-.3-.3-.3-.7-.2-1.1L7.5 5.3V9c.1 0 .2.1.3.2.4.4.4 1.1 0 1.5s-1.1.4-1.5 0-.4-1.1 0-1.5c.1-.1.2-.2.3-.2V5.2c-.1-.1-.2-.1-.3-.2-.3-.4-.4-.8-.2-1.2L4.5 2.3.4 6.4c-.3.3-.3.9 0 1.2l6 6c.3.3.9.3 1.2 0l6-6c.3-.3.3-.9 0-1.2z"
                                    fill="#f42534"
                                  />
                                </svg>
                              </span>
                            </div>
                            <p
                              class="StyledProps--font StyledProps--main Thumbnail--label StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey600 StyledProps--margin-top-small"
                            >
                              Git
                            </p>
                            <input
                              checked=""
                              name="store"
                              type="checkbox"
                              value="Git"
                            />
                          </label>
                          <button
                            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main ThumbnailSelect--changeButton StyledProps--intent-primary Button--without-shadow Button--withLeftIcon"
                            data-testid="thumbnail-select-change"
                            type="button"
                          >
                            <span
                              class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
                              data-icon="Edit"
                            >
                              <svg
                                height="12"
                                viewBox="0 0 16 16"
                                width="12"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  clip-rule="evenodd"
                                  d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                />
                                <path
                                  clip-rule="evenodd"
                                  d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                  opacity="0.3"
                                />
                              </svg>
                            </span>
                            <span
                              class="bp3-button-text"
                            >
                              Change
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main connectorContainer StyledProps--flex StyledProps--flex-alignItems-flex-start StyledProps--flex-justifyContent-flex-start"
                  >
                    <div
                      class="connectorLabel"
                    >
                      <div
                        class="StyledProps--font StyledProps--main"
                        style="margin-bottom: 5px;"
                      >
                        <span
                          class="Tooltip--acenter bp3-label"
                          data-tooltip-id="manifestStore_connectorRef"
                        >
                          Git connector
                        </span>
                      </div>
                      <div
                        class="bp3-form-group"
                        data-id="undefined-1"
                        style="margin-bottom: 0px;"
                      >
                        <div
                          class="bp3-form-content"
                        >
                          <div
                            class="Layout--horizontal StyledProps--main MultiTypeInput--main"
                            style="width: 400px;"
                          >
                            <button
                              class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main container Button--iconOnly Button--without-shadow Button--withRightIcon"
                              data-testid="cr-field-connectorRef"
                              style="width: 400px;"
                              type="button"
                            >
                              <span
                                class="bp3-button-text"
                              >
                                <span
                                  class="placeholder"
                                >
                                  Git connector
                                </span>
                              </span>
                              <span
                                class="bp3-icon bp3-icon-chevron-down StyledProps--main"
                                icon="chevron-down"
                              >
                                <svg
                                  data-icon="chevron-down"
                                  height="14"
                                  viewBox="0 0 16 16"
                                  width="14"
                                >
                                  <desc>
                                    chevron-down
                                  </desc>
                                  <path
                                    d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                                    fill-rule="evenodd"
                                  />
                                </svg>
                              </span>
                            </button>
                            <span
                              class="bp3-popover-wrapper MultiTypeInput--wrapper"
                            >
                              <span
                                class="bp3-popover-target"
                              >
                                <button
                                  class="MultiTypeInput--btn MultiTypeInput--FIXED"
                                >
                                  <span
                                    class="bp3-icon StyledProps--main"
                                    data-icon="fixed-input"
                                  >
                                    <svg
                                      height="12"
                                      viewBox="0 0 9 9"
                                      width="12"
                                      xmlns="http://www.w3.org/2000/svg"
                                    >
                                      <path
                                        d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                                        fill="currentColor"
                                      />
                                    </svg>
                                  </span>
                                </button>
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                    <button
                      class="bp3-button Button--button StyledProps--font StyledProps--main addNewManifest Button--with-current-color Button--withLeftIcon Button--variation Button--variation-link Button--size Button--small"
                      id="new-releaserepo-connector"
                      type="button"
                    >
                      <span
                        class="bp3-icon bp3-icon-plus StyledProps--main StyledProps--padding-right-xsmall"
                        icon="plus"
                      >
                        <svg
                          data-icon="plus"
                          height="12"
                          viewBox="0 0 16 16"
                          width="12"
                        >
                          <desc>
                            plus
                          </desc>
                          <path
                            d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                      <span
                        class="bp3-button-text"
                      >
                        newLabel Git connector
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main saveBtn"
                >
                  <button
                    class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Button--with-current-color Button--withRightIcon Button--variation Button--variation-primary"
                    disabled=""
                    tabindex="-1"
                    type="submit"
                  >
                    <span
                      class="bp3-button-text"
                    >
                      continue
                    </span>
                    <span
                      class="bp3-icon bp3-icon-chevron-right StyledProps--main"
                      icon="chevron-right"
                    >
                      <svg
                        data-icon="chevron-right"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                      >
                        <desc>
                          chevron-right
                        </desc>
                        <path
                          d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
    <span
      class="bp3-icon StyledProps--main StepWizard--harnessWatermark StyledProps--color StyledProps--color-grey50"
      data-icon="harness-with-color"
    >
      <svg
        height="346"
        viewBox="0 0 25 25"
        width="346"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M22.974 8.186L16.708 1.88a7.023 7.023 0 00-2.59-1.572C11.98-.409 9.858.154 8.135 1.88L1.868 8.186a7.081 7.081 0 00-1.562 2.606c-.713 2.153-.153 4.289 1.562 6.023l6.266 6.306a7.113 7.113 0 002.589 1.58 5.663 5.663 0 001.808.308c1.503 0 2.946-.64 4.177-1.88l6.266-6.305a7.08 7.08 0 001.562-2.606c.713-2.162.153-4.298-1.562-6.032zm-10.07-4.477c.706.213 1.232.7 1.274.735l1.851 1.862-3.608 3.631-3.608-3.63 1.859-1.872c.5-.496 1.197-1.034 2.233-.726zm-9.219 8.305c.212-.71.696-1.24.73-1.282l1.85-1.863 3.609 3.632-3.608 3.63-1.86-1.87c-.492-.504-1.027-1.214-.721-2.247zm8.252 9.279c-.705-.214-1.231-.701-1.273-.735l-1.851-1.863 3.608-3.631 3.608 3.631-1.86 1.871c-.5.496-1.205 1.034-2.232.726zm9.22-8.305c-.212.709-.696 1.239-.73 1.281l-1.851 1.863-3.608-3.631 3.608-3.632 1.86 1.872c.492.504 1.027 1.204.72 2.247z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`Release Repo wizard tests on edit 1`] = `
<div>
  <div
    class="StepWizard--main manifestWizard StepWizard--navBar"
  >
    <div
      class="StepWizard--navBarList"
    >
      <div
        class="StepWizard--navStep StepWizard--activeStep"
      >
        <span
          class="StepWizard--number"
        >
          1
        </span>
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main StepWizard--stepName"
          style="--text-line-clamp: 2; width: 200px;"
        >
          Release Repo Store
        </p>
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main StepWizard--stepName"
          style="--text-line-clamp: 2; width: 200px;"
        >
          
        </p>
      </div>
      <div
        class="StepWizard--navStep"
      >
        <span
          class="StepWizard--number"
        >
          2
        </span>
        <p
          class="StyledProps--font Text--lineclamp StyledProps--main StepWizard--stepName"
          style="--text-line-clamp: 2; width: 200px;"
        >
          RepoDetails
        </p>
      </div>
    </div>
    <div
      class="StepWizard--stepDetails"
    >
       
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main optionsViewContainer"
        style="height: inherit;"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-variation-h3 StyledProps--margin-bottom-medium"
        >
          Release Repo Store
        </p>
        <div
          class="OverlaySpinner--overlaySpinner"
        >
          <form
            action="#"
            class="FormikForm--main"
          >
            <div>
              <div
                class="Layout--vertical StyledProps--main manifestForm StyledProps--flex StyledProps--flex-justifyContent-space-between StyledProps--flex-alignItems-flex-start"
              >
                <div
                  class="Layout--vertical StyledProps--main"
                >
                  <div
                    class="Layout--horizontal Layout--layout-spacing-large StyledProps--main"
                  >
                    <div
                      class="bp3-form-group ThumbnailSelect--mainContainer thumbnailSelect"
                      data-id="store-0"
                    >
                      <div
                        class="bp3-form-content"
                      >
                        <div
                          class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
                        >
                          <label
                            class="Thumbnail--squareCardContainer"
                          >
                            <div
                              class="bp3-card bp3-elevation-0 Card--card Thumbnail--squareCard Card--selected"
                            >
                              <span
                                class="Card--corner"
                              >
                                <span
                                  class="Card--badge"
                                />
                                <span
                                  class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-white"
                                  data-icon="main-tick"
                                >
                                  <svg
                                    height="10"
                                    viewBox="0 0 11 8"
                                    width="10"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M9.219.894c.356-.38.982-.413 1.398-.072.391.32.457.857.167 1.236l-.06.07-5.138 5.486c-.362.386-.993.41-1.403.068l-.07-.064-3.2-3.214A.88.88 0 01.95 3.098a1.045 1.045 0 011.328-.003l.074.067 2.443 2.454L9.22.894z"
                                      fill="currentColor"
                                    />
                                  </svg>
                                </span>
                              </span>
                              <span
                                class="bp3-icon StyledProps--main"
                                data-icon="service-github"
                              >
                                <svg
                                  height="26"
                                  viewBox="0 0 14 14"
                                  width="26"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M13.6 6.4l-6-6C7.3.1 6.7.1 6.4.4L5.1 1.6l1.6 1.6c.4-.1.8 0 1.1.3s.4.7.2 1.1l1.5 1.5c.4-.1.8 0 1.1.2.4.4.4 1.1 0 1.5s-1.1.4-1.5 0c-.3-.3-.3-.7-.2-1.1L7.5 5.3V9c.1 0 .2.1.3.2.4.4.4 1.1 0 1.5s-1.1.4-1.5 0-.4-1.1 0-1.5c.1-.1.2-.2.3-.2V5.2c-.1-.1-.2-.1-.3-.2-.3-.4-.4-.8-.2-1.2L4.5 2.3.4 6.4c-.3.3-.3.9 0 1.2l6 6c.3.3.9.3 1.2 0l6-6c.3-.3.3-.9 0-1.2z"
                                    fill="#f42534"
                                  />
                                </svg>
                              </span>
                            </div>
                            <p
                              class="StyledProps--font StyledProps--main Thumbnail--label StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey600 StyledProps--margin-top-small"
                            >
                              Git
                            </p>
                            <input
                              checked=""
                              name="store"
                              type="checkbox"
                              value="Git"
                            />
                          </label>
                          <button
                            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main ThumbnailSelect--changeButton StyledProps--intent-primary Button--without-shadow Button--withLeftIcon"
                            data-testid="thumbnail-select-change"
                            type="button"
                          >
                            <span
                              class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
                              data-icon="Edit"
                            >
                              <svg
                                height="12"
                                viewBox="0 0 16 16"
                                width="12"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  clip-rule="evenodd"
                                  d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                />
                                <path
                                  clip-rule="evenodd"
                                  d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                  opacity="0.3"
                                />
                              </svg>
                            </span>
                            <span
                              class="bp3-button-text"
                            >
                              Change
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main connectorContainer StyledProps--flex StyledProps--flex-alignItems-flex-start StyledProps--flex-justifyContent-flex-start"
                  >
                    <div
                      class="connectorLabel"
                    >
                      <div
                        class="StyledProps--font StyledProps--main"
                        style="margin-bottom: 5px;"
                      >
                        <span
                          class="Tooltip--acenter bp3-label"
                          data-tooltip-id="manifestStore_connectorRef"
                        >
                          Git connector
                        </span>
                      </div>
                      <div
                        class="bp3-form-group"
                        data-id="undefined-1"
                        style="margin-bottom: 0px;"
                      >
                        <div
                          class="bp3-form-content"
                        >
                          <div
                            class="Layout--horizontal StyledProps--main MultiTypeInput--main"
                            style="width: 400px;"
                          >
                            <button
                              class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main container Button--iconOnly Button--without-shadow Button--withRightIcon"
                              data-testid="cr-field-connectorRef"
                              style="width: 400px;"
                              type="button"
                            >
                              <span
                                class="bp3-button-text"
                              >
                                <div
                                  class="Layout--horizontal Layout--layout-spacing-small StyledProps--main selectWrapper StyledProps--flex StyledProps--flex-distribution-space-between"
                                >
                                  <p
                                    class="StyledProps--font StyledProps--main label StyledProps--color StyledProps--color-grey800"
                                  />
                                  <div
                                    class="rightStatus"
                                  >
                                    <span
                                      class="bp3-icon bp3-icon-full-circle StyledProps--main status redStatus"
                                      data-testid="crf-status"
                                      icon="full-circle"
                                    >
                                      <svg
                                        data-icon="full-circle"
                                        height="6"
                                        viewBox="0 0 16 16"
                                        width="6"
                                      >
                                        <desc>
                                          full-circle
                                        </desc>
                                        <path
                                          d="M8 0a8 8 0 100 16A8 8 0 108 0z"
                                          fill-rule="evenodd"
                                        />
                                      </svg>
                                    </span>
                                    <span
                                      class="Tag--main"
                                    >
                                      <span
                                        class="bp3-tag bp3-minimal"
                                        id="tag"
                                      >
                                        <span
                                          class="bp3-text-overflow-ellipsis bp3-fill"
                                        >
                                          project
                                        </span>
                                      </span>
                                    </span>
                                  </div>
                                </div>
                              </span>
                              <span
                                class="bp3-icon bp3-icon-chevron-down StyledProps--main"
                                icon="chevron-down"
                              >
                                <svg
                                  data-icon="chevron-down"
                                  height="14"
                                  viewBox="0 0 16 16"
                                  width="14"
                                >
                                  <desc>
                                    chevron-down
                                  </desc>
                                  <path
                                    d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                                    fill-rule="evenodd"
                                  />
                                </svg>
                              </span>
                            </button>
                            <span
                              class="bp3-popover-wrapper MultiTypeInput--wrapper"
                            >
                              <span
                                class="bp3-popover-target"
                              >
                                <button
                                  class="MultiTypeInput--btn MultiTypeInput--FIXED"
                                >
                                  <span
                                    class="bp3-icon StyledProps--main"
                                    data-icon="fixed-input"
                                  >
                                    <svg
                                      height="12"
                                      viewBox="0 0 9 9"
                                      width="12"
                                      xmlns="http://www.w3.org/2000/svg"
                                    >
                                      <path
                                        d="M5.61 7.55c.62-.7.3-1.42.18-2.35a.32.32 0 01.1-.29l1.43-1.42a.76.76 0 01.28-.15c.39-.1 1 0 1.28-.29a.34.34 0 000-.51L6.46.12a.33.33 0 00-.52 0c-.28.29-.2.9-.28 1.3a.53.53 0 01-.13.23L4.09 3.12a.29.29 0 01-.28.09c-.89-.11-1.6-.49-2.27.11-.29.26-.29.43 0 .7l1.31 1.34.15.16-.15.15-2.7 2.69a.36.36 0 000 .54.36.36 0 00.54-.05l2.63-2.68.17-.17.17.16c.45.46.9.91 1.36 1.36.21.21.42.22.59.03z"
                                        fill="currentColor"
                                      />
                                    </svg>
                                  </span>
                                </button>
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                    <button
                      class="bp3-button Button--button StyledProps--font StyledProps--main addNewManifest Button--with-current-color Button--withLeftIcon Button--variation Button--variation-link Button--size Button--small"
                      id="new-releaserepo-connector"
                      type="button"
                    >
                      <span
                        class="bp3-icon bp3-icon-plus StyledProps--main StyledProps--padding-right-xsmall"
                        icon="plus"
                      >
                        <svg
                          data-icon="plus"
                          height="12"
                          viewBox="0 0 16 16"
                          width="12"
                        >
                          <desc>
                            plus
                          </desc>
                          <path
                            d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                      <span
                        class="bp3-button-text"
                      >
                        newLabel Git connector
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main saveBtn"
                >
                  <button
                    class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Button--with-current-color Button--withRightIcon Button--variation Button--variation-primary"
                    disabled=""
                    tabindex="-1"
                    type="submit"
                  >
                    <span
                      class="bp3-button-text"
                    >
                      continue
                    </span>
                    <span
                      class="bp3-icon bp3-icon-chevron-right StyledProps--main"
                      icon="chevron-right"
                    >
                      <svg
                        data-icon="chevron-right"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                      >
                        <desc>
                          chevron-right
                        </desc>
                        <path
                          d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
    <span
      class="bp3-icon StyledProps--main StepWizard--harnessWatermark StyledProps--color StyledProps--color-grey50"
      data-icon="harness-with-color"
    >
      <svg
        height="346"
        viewBox="0 0 25 25"
        width="346"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M22.974 8.186L16.708 1.88a7.023 7.023 0 00-2.59-1.572C11.98-.409 9.858.154 8.135 1.88L1.868 8.186a7.081 7.081 0 00-1.562 2.606c-.713 2.153-.153 4.289 1.562 6.023l6.266 6.306a7.113 7.113 0 002.589 1.58 5.663 5.663 0 001.808.308c1.503 0 2.946-.64 4.177-1.88l6.266-6.305a7.08 7.08 0 001.562-2.606c.713-2.162.153-4.298-1.562-6.032zm-10.07-4.477c.706.213 1.232.7 1.274.735l1.851 1.862-3.608 3.631-3.608-3.63 1.859-1.872c.5-.496 1.197-1.034 2.233-.726zm-9.219 8.305c.212-.71.696-1.24.73-1.282l1.85-1.863 3.609 3.632-3.608 3.63-1.86-1.87c-.492-.504-1.027-1.214-.721-2.247zm8.252 9.279c-.705-.214-1.231-.701-1.273-.735l-1.851-1.863 3.608-3.631 3.608 3.631-1.86 1.871c-.5.496-1.205 1.034-2.232.726zm9.22-8.305c-.212.709-.696 1.239-.73 1.281l-1.851 1.863-3.608-3.631 3.608-3.632 1.86 1.872c.492.504 1.027 1.204.72 2.247z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
</div>
`;
